src/types/sgrStack.cpp

```diff
@@ -19,42 +19,31 @@ namespace Microsoft::Console::VirtualTerminal
     {
         AttrBitset validParts;
 
-        try
+        if (options.empty())
         {
-            if (options.empty())
-            {
-                // We save all current attributes.
-                validParts.set(static_cast<size_t>(SgrSaveRestoreStackOptions::All));
-            }
-            else
+            // We save all current attributes.
+            validParts.set(SgrSaveRestoreStackOptions::All);
+        }
+        else
+        {
+            // Each option is encoded as a bit in validParts. All options (that fit) are
+            // encoded; options that aren't supported are ignored when read back (popped).
+            // So if you try to save only unsupported aspects of the current text
+            // attributes, you'll do what is effectively an "empty" push (the subsequent
+            // pop will not change the current attributes), which is the correct behavior.
+
+            for (size_t i = 0; i < options.size(); i++)
             {
-                // Each option is encoded as a bit in validParts. All options (that fit) are
-                // encoded; options that aren't supported are ignored when read back (popped).
-                // So if you try to save only unsupported aspects of the current text
-                // attributes, you'll do what is effectively an "empty" push (the subsequent
-                // pop will not change the current attributes), which is the correct behavior.
+                const auto option = static_cast<SgrSaveRestoreStackOptions>(options.at(i).value_or(0));
 
-                for (size_t i = 0; i < options.size(); i++)
+                // Options must be specified singly; not in combination. Values that are
+                // out of range will be ignored.
+                if (option > SgrSaveRestoreStackOptions::All && option <= SgrSaveRestoreStackOptions::Max)

```

Comment:
Shouldn't this be `option >= All && option < Max`?

Reply:
Oh `Max` is actually inclusive... I thought it's like the typical exclusive "Count" enum member that's 1 past the highest element. But what about `All`? You can't set the option to 0 (aka `All`) via VT?

Reply:
> if an option is 0 (All), it will now just be ignored.

Fuck. 😄 I need to learn to read. 😄